Fixes issue with Installed Tab when the machine is offline

deepakaravindr opened the pull request:
Fixes https://github.com/NuGet/Home/issues/1364

Manually tested that the Installed Filter does not show an error,
when there is a fake package source or if the machine is offline

@yishaigalatzer @emgarten @feiling @MeniZalzman

deepakaravindr commented:
With the fix, 'Installed' filter in offline or bad package source will look like the following

![image](https://cloud.githubusercontent.com/assets/2380340/9779398/eeaf6cd2-572f-11e5-889f-e94b69cab860.png)

yishaigalatzer commented on src/PackageManagement.UI/PackageLoader.cs:
```diff
@@ -226,10 +226,22 @@ private IEnumerable<string> GetSupportedFrameworks()
                 null :
                 await _sourceRepository.GetResourceAsync<UIMetadataResource>();
+            }
+            catch (Exception)
```
Don't catch Exception. Either do an empty catch, or catch the specific exception we want to get.

Also please open up a follow-up issue so we can clean this up nicely and do the right thing for the description and the icons on the package (because it shouldn't show check if the package *might have an update, instead it should show no icons).

emgarten replied:
We should consider wrapping the service index exceptions in a protocol exception to make this easier to catch. This same issue likely exists in many other scenarios.

deepakaravindr committed:
Fixes https://github.com/NuGet/Home/issues/1364

Manually tested that the Installed Filter does not show an error,
when there is a fake package source or if the machine is offline

emgarten commented:
:rocket:

MeniZalzman commented:
:shipit:
